### ISSUE TITLE
Cannot delete sites from "Sites that can always use cookies"

### ISSUE THREAD

**Issue by aperullo** [brave/brave-browser#12375]
## Description 
Sites cannot be deleted from the "always allow cookies" list, only added

## Steps to Reproduce

   1. Add a site to "Sites that can always use cookies"
   2. Try to click the "garbage" icon to delete a site
   3. Nothing will happen

## Actual result:
Nothing occurs and the site remains

## Expected result:
The site should be deleted

## Reproduces how often: 
Consistent

## Brave version (brave://version info)
1.16.68 Chromium: 86.0.4240.111 (Official Build) (64-bit) b8c36128a06ebad76af51591bfec980224db5522-refs/branch-heads/4240@{#1290}

## OS
Windows 10 64-bit

**Comment by rebron** [comment]
cc: @jumde Related to the google login exceptions?

**Comment by jumde** [comment]
@aperullo - Are you still seeing this if `Allow Google login buttons on third party sites` is disabled in brave://settings/socialBlocking

**Comment by aperullo** [comment]
@jumde Yeah, the issue persists when that setting is toggled off or on. Even after restarting the browser. One of the stuck sites is almost certainly not using Google login either, as its a microsoft domain. Please let me know how else I can help.

**Comment by dentistformyeye** [comment]
Seems to be a duplicate of #11259

**Comment by optimistiCli** [comment]
Same behaviour on Linux  1.16.68 x86_64

**Comment by elonj** [comment]
Same behaviour on Mac 1.16.72 Chromium: 86.0.4240.183 (Official Build) (x86_64)

**Comment by 67no** [comment]
Similar issue here Version 1.16.72 Chromium: 86.0.4240.183 (Offizieller Build) (64-Bit). There are three entries in the "Sites that can always use cookies" section that cannot be removed. Even worse, they allow third party cookies which I have **never** done. Whenever I manually ad entries I either used the cookies menu in the menu that pops up when you clock the lock icon to the left of the url bar (which has no way of enabling all third party cookies) or make sure the check box is unchecked.

**Comment by elonj** [comment]
Same behaviour on Version 1.17.73 Chromium: 87.0.4280.67 (Official Build) (64-bit)

**Comment by rotatingangles** [comment]
**SOLUTION** on Brave Browser Linux.
Go to website in question and toggle the "Shields" to "UP" as opposed to "DOWN"  
Try it on github, you'll see it added or removed.
This is one hell of a way for this to operate.
Brave, give the user some control over this, make it more user friendly, and absolutely make it more easily removable.
The default DOWN should be temporary, maybe clear cookies when windows are closed.
There should be an UP, DOWN, etc., maybe a WTF as I can't believe this polar thinking!

**Comment by rotatingangles** [comment]
**SOLUTION** -Go to website in question, and change Shields to UP.

**Comment by bsclifton** [comment]
cc: @rebron @karenkliu

**Comment by snowbound** [comment]
This is still happening on Version 1.18.78 Chromium: 87.0.4280.141 (Official Build) (64-bit) and Version 1.19.77 Chromium: 87.0.4280.101 (Official Build) beta (64-bit) on Windows10 x64 20H2. As well if you add google.com to the list of sites under Always clear cookies when windows are closed the cookie is not being deleted. If it was then one would be logged out of Gmail.

This deletion of Google.com sub cookies works fine in MS Edge Chromium and Google Chrome.

**Comment by alexfrederiksen** [comment]
This is still happening for me as well on Arch Linux 5.10.7-arch1-1 with brave-bin 1:1.19.86-1. It happens in the "always allow" and "never allow" sections.

**Comment by Izofeu** [comment]
Still a problem in Version 1.19.92 Chromium: 88.0.4324.152 (Official Build) (64-bit)

**Comment by snowbound** [comment]
I am also tagging this issue https://github.com/brave/brave-browser/issues/9085 as it too is cookie deletion related but on exit of the browser.

**Comment by gloist** [comment]
> @aperullo - Are you still seeing this if `Allow Google login buttons on third party sites` is disabled in brave://settings/socialBlocking

didn't work.

**Comment by rosolam** [comment]
I have sites that I cannot delete in both **Sites that can always use cookies** and **Sites that can never use cookies** and interestingly I have one stuck in both that is preventing me from using the website in Brave.  Any way to adjust these settings from outside of brave?

**Comment by Izofeu** [comment]
> I have sites that I cannot delete in both **Sites that can always use cookies** and **Sites that can never use cookies** and interestingly I have one stuck in both that is preventing me from using the website in Brave. Any way to adjust these settings from outside of brave?



> Still a problem in Version 1.19.92 Chromium: 88.0.4324.152 (Official Build) (64-bit)
> Edit: I found a workaround to delete those entries - go to \AppData\Local\BraveSoftware\Brave-Browser\User Data\Default. Open file called Preferences and delete all site entries that you want to be gone. For example, if you want to delete a cookie permission from site "xxx.com", delete the following from the file (make sure the browser is closed):
> `"xxx.com,*":{"expiration":"0","last_modified":"<somerandomnumbers>","model":0,"setting":2},`
> Edit2: For sites that say "embedded on xxx.com" you might need to delete additional lines that look similar to the one above.

**Comment by nisc** [comment]
Hey guys, can confirm that has been a problem on Mac OS and Windows for a while. Here is the thread in the BraveCommunity Forums and my (inconvenient) workaround:

https://community.brave.com/t/cant-remove-sites-in-settings-sites-that-can-always-use-cookies/176438/13?u=nisc

As I describe there, the Sync Chain can be a big part of the problem. Even after deleting the entries from the `Preferences` file, Brave would auto-restore the settings immediately. I had to create a new Sync Chain AND clean up the `Preferences` file to resolve the issue.

[quote="nisc, post:13, topic:176438"]
Okay I spent some time investigating this issue and I believe it’s a bug in the Brave Sync Chain mechanism. I grepped through my whole Brave profile directory and investigated all references to the websites for which cookie settings could no longer be deleted.

**The only way I found to really fix this issue is to remove all my devices from the current sync chain, then thoroughly clean up the `Preferences` JSON file in my Profile directory (see below), remove ( `rm -rf` ) all the the sync caches ( `Brave-Browser/Default/Sync Data/LevelDB` and `Brave-Browser/Default/Service Worker/CacheStorage` ), and then finally create a new Sync chain and rejoin from all my devices.**

I used a RegEx pattern similar to the below to clean up my `Preferences` file:
`%s/((?<=\":\{)|[\,])\"(https:\/\/)?(voice|mail|calendar|docs|drive|world|www)\.(slideshare|hyatt|google)\.(com|net)[^\{]+\{[^\}]+\}[\,]?//g`

*(Please be careful with your pattern … if you delete just one wrong character, Brave marks the file as “bad” and overwrites most of it. If you’re not into RegEx, you could also try clearing your Site Settings after leaving the Sync chain.)*

If you don’t want to recreate your Sync chain for whatever reason, the only other way to work around this issue is to disable the “Settings” synchronization in `brave://settings/braveSync/setup` before cleaning up the `Preferences` file as described above. You would no longer have settings synchronization, though.

Good luck.
[/quote]

**Comment by dentistformyeye** [comment]
>Okay I spent some time investigating this issue and I believe it’s a bug in the Brave Sync Chain mechanism.

I've visited the sync setup page (brave://settings/braveSync/setup) and have pressed some of the buttons on that page before, but I've never set up a sync chain, so I'm unsure that the sync mechanism is the cause of this bug.

**Comment by snowbound** [comment]
First I want to apologize for the following post. Its a product of my frustration with Brave at times and bugs that only Brave seems to have amongst Chromium browsers that I have used.

For the past couple of days, I have switched to MS Edge Chromium on multiple Windows computers and on iOS. I have not experienced either this issue nor https://github.com/brave/brave-browser/issues/11183 nor https://github.com/brave/brave-browser/issues/9085.

 Both Brave and MS Edge are chromium-based browsers. Only Brave uses a passphrase to allow a sync chain. With Brave even recreating a sync chain from scratch does not help solve any of these issues.

**Comment by nisc** [comment]
> > Okay I spent some time investigating this issue and I believe it’s a bug in the Brave Sync Chain mechanism.
> 
> I've visited the sync setup page (brave://settings/braveSync/setup) and have pressed some of the buttons on that page before, but I've never finished setting up a sync chain, so I'm unsure that the sync mechanism is the cause of this bug.

I also don't think that this is the cause of the funny cookie settings entries, but it is probably what makes them "Zombie Cookie" like after they first appear. Otherwise I could just clear my site settings and the funny entries should be gone.

**Comment by Brave-Matt** [comment]
@rebron can we get an update on this issue?

**Comment by Michael82548** [comment]
Affects me as well. This can be a potential security risk if someone wants to prevent website cross tracking but can't because they can't remove permissions

**Comment by Michael82548** [comment]
I'm honestly surprised that brave also does not allow cookie exceptions for SPECIFIC websites (Using third party cookies). Not to apply globally like it normally does. This is extremely annoying. Except now, I can't even remove them.
![image](https://i.imgur.com/9z5vTXC.gif)

**Comment by Izofeu** [comment]
> I'm honestly surprised that brave also does not allow cookie exceptions for SPECIFIC websites (Using third party cookies). Not to apply globally like it normally does. This is extremely annoying. Except now, I can't even remove them. Found a GIF in this thread: https://community.brave.com/t/cant-remove-sites-in-settings-sites-that-can-always-use-cookies/176438
> ![image](https://camo.githubusercontent.com/00a9189f88a48514f7cd552a788bfd75fa5f1ad26260650a46d78fc059b17bf8/68747470733a2f2f692e696d6775722e636f6d2f397a35765458432e676966)

Brave allows cookie exceptions for specific sites. If you want to allow 3rd party cookies on one site, change it in Brave Shield settings.
![image](https://user-images.githubusercontent.com/60717460/116628766-1fbeae00-a950-11eb-87c6-d71e2702a0e2.png)

**Comment by snowbound** [comment]
Brave will not even delete specific cookies on the closing of the browser and has not done so for over a frustrating year https://github.com/brave/brave-browser/issues/9085 I have given up on these items ver getting fixed or Sync v2 ever working properly. Other Chromium based browsers do not have these issues and it is darn frustrating to hear excuses after excuses for a browser whose marketing spiel is security and privacy  yet these types of issues remain unfixed.

I guess adding features and looking after BAT issues are more important to Brave.

**Comment by fusionneur** [comment]
Same for me on Version 1.23.75 Chromium: 90.0.4430.93 (Official Build) (64-bit)

**Comment by Michael82548** [comment]
I have Cross-site cookies blocked but it does not allow me to make exceptions for specific sites. Also, the chromium cookie settings override the Brave settings

**Comment by M09NST3R** [comment]
I have 11 pages/screens (!) of Cookie settings that I cannot delete. It seems that the more you change the default Shield Setting (I do it on at least half of sites I visit), the more of these Zombie settings accumulate over time.

I don't understand how much of this is just an UI annoyance/artifact vs. actually ignored user-set preferences. However, fact is that Brave does something without the user understanding what it is.

@rebron @Brave-Matt

**Comment by nisc** [comment]
Still the same problem in V1.23.75 and it's getting worse every day.

@rebron @jumde @Brave-Matt

**Comment by Michael82548** [comment]
This affects Android also (in sync with desktop (Windows))

The way it works on mobile is that you can block cookies that were previously allowed but when you delete blocked cookies, it moves it back to "allowed" and when you try to remove them from allowed, it does nothing.

**Comment by Michael82548** [comment]
Add OS/Android @Brave-Matt

**Comment by yeldarb1983** [comment]
okay, just adding some stuff in the hopes of it being useful to tracking this down.

1) I've NEVER set up sync, nor do I intend to. I'm sure it's a fine feature if you want it, but it's really not that useful to me.

2) How are these sites being set to "always allow" in the first place? I've NEVER added a website to my  whitelist, yet a dozen or so have exceptions, which tells me either there's some code in the browser itself that does it automatically, which should have an "always ask" option or some other mechanism(see next point)

3) I'm wondering if this isn't some bug in the code that certain companies have exploited to force their cookies to remain regardless of user interaction. It may even be present in other browsers based on chromium, but hidden from the user, and something that brave did just happened to make it visible to the user?

hoping this is useful.

**Comment by M09NST3R** [comment]
> 3\. I'm wondering if this isn't some bug in the code that certain companies have exploited to force their cookies to remain regardless of user interaction. It may even be present in other browsers based on chromium, but hidden from the user, and something that brave did just happened to make it visible to the user?

I thought this initially, but then I also found several reputable / privacy-oriented sites in my zombie exceptions list ... thus, I believe that's not the case here.

**Comment by drstevens** [comment]
Changing brave settings through the lion icon once a page has rendered does appear to work in most cases for me. It's tedious though, and hard to do when the domain in question causes 302 redirects to other domains.

**Comment by nisc** [comment]
Unless you use Brave with very permissive default Shield cookie settings (cookies allowed or at least same-site cookies allowed), the domain-level exceptions site (brave://settings/cookies) will become unusably polluted with these zombie cookies very quickly. Not a good UX.

**Comment by Michael82548** [comment]
It looks like I can edit the cookie URL, then it saves, then when I try and delete it, it reverts back to the initial cookie

**Comment by Michael82548** [comment]
![image](https://user-images.githubusercontent.com/53535277/117083666-3465c200-ad13-11eb-81d6-bb9a694febb4.png)
It just multiplies. I can delete these entries fine but not the zombie cookies

**Comment by Michael82548** [comment]
Clarifying, If I edit a zombie cookie that I moved to the "block" list, then it creates a new entry on the blocklist and moves the cookie back to the allow list

**Comment by Michael82548** [comment]
It looks like it is only visually in the block list. It is always in the allow list.

**Comment by Michael82548** [comment]
> Brave allows cookie exceptions for specific sites. If you want to allow 3rd party cookies on one site, change it in Brave Shield settings.
> ![image](https://user-images.githubusercontent.com/60717460/116628766-1fbeae00-a950-11eb-87c6-d71e2702a0e2.png)

BTW Brave's "Cross site cookies blocked" option in shields seems to do nothing when you are affected by this bug.

Test it yourself https://browserleaks.com/social

Its leaking for me and it also shows in the cookies menu despite cross site cookies being disabled
![image](https://user-images.githubusercontent.com/53535277/117084308-de921980-ad14-11eb-9e03-22a9f7d941cc.png)
![image](https://user-images.githubusercontent.com/53535277/117084324-ea7ddb80-ad14-11eb-82cd-c6e50abba71d.png)

**Comment by Michael82548** [comment]
If you want to make a browser that blocks "Tracking" (Google), at least google chrome allows us to make these exceptions!

The only thing that I can POSSIBLY do right now is nuke my profile. Worst part about that is the fact that the bug isn't fixed so it will just continue happening again. I am sick of this.

**Comment by Michael82548** [comment]
I simply cannot recommend this browser due to this potential security issue that people probably won't even know about! This doesn't just affect "power users"

**Comment by snowbound** [comment]
If you use multiple Gmail accounts there is no easy way to force a logout of all accounts by simply closing the browser and deleting one cookie. You have to do a scorch earth deletion of all google.com cookies which will wipe out all configuration for YouTube and 2FA cookies that you want to keep. This has been a festering problem I have been battling with Brave for over a year now.

**Comment by nisc** [comment]
Brian in da house. Seems we're finally getting some attention here. Can we move this to P1 maybe?

**Comment by Michael82548** [comment]
Please...

**Comment by Michael82548** [comment]
Also add Android tag. It seems to affect android as well

**Comment by pitsi** [comment]
Same thing in 1.24.84 and 1.24.85.

**Comment by lllusion303** [comment]
Confirmed bug in 1.24.86 Chromium: 90.0.4430.212 (Official Build) (arm64) on macOS 11.3.1.
Cannot delete cookies from "Sites that can always use cookies". I've not allowed 99% of the sites listed under this heading.

**Comment by Michael82548** [comment]
> Confirmed bug in 1.24.86 Chromium: 90.0.4430.212 (Official Build) (arm64) on macOS 11.3.1.
> Cannot delete cookies from "Sites that can always use cookies". I've not allowed 99% of the sites listed under this heading.
> 
> Creating a new profile automatically places two cookies in the "always" list.
> <img alt="Screenshot 2021-05-24 at 19 19 07" width="676" src="https://user-images.githubusercontent.com/84734835/119384401-f120c280-bcc4-11eb-9336-f551bdd8b0e1.png">

I also experience this issue. Version 1.26.34 Chromium: 91.0.4472.57 (Official Build) beta (64-bit)

I loaded some test websites with a new profile and whenever I allow cookies or change a shield setting in brave (Such as fingerprinting), some of these cookies are stuck now. I cannot remove them.

**Comment by Michael82548** [comment]
If you cannot remove sensitive cookies from the always allowed list, then it is literally a security flaw. Any website to access them, even if 3rd party cookies are enabled.

**Comment by nisc** [comment]
> literally a security flaw.

This. I don't understand what the F is going on. This has been raised more than half a year ago, has a huge thread on their community forums, the team has been pinged many times by several concerned users.

But it's still a "P3" issue in Github and there is no communication from the team at all.

How can they not take this seriously? @bsclifton

**Comment by nisc** [comment]
> But it's still a "P3" issue in Github and there is no communication from the team at all.

**Label descriptions:**
P3 — The next thing for us to work on. It'll ride the trains. (454 open issues and pull requests)
P2 — A bad problem. We might uplift this to the next planned release. (128 open issues and pull requests)
P1 — A very extremely bad problem. We might push a hotfix for it. (4 open issues and pull requests)

🤣

**Comment by snowbound** [comment]
Don't forget about this other cookie deletion issue on Exit of Brave that still does not work properly https://github.com/brave/brave-browser/issues/9085 you have to resort to scorch earth and delete all cookies.

**Comment by Michael82548** [comment]
If Google is ditching cookies for FLoC then Brave should be Ditching FLoC WHILE making sure that cookie management is still functional. @snowbound

**Comment by Michael82548** [comment]
The fact that this is labeled as P2 is giving me some hope now. It still seems to be in the P3 backlog though... Probably just not being worked on at the moment :/

**Comment by snowbound** [comment]
@Michael82548 Well as you can see by my original bug report on google.com cookie that it was reported well over a year ago and only after I battled w the browser and Brave people in Brave community.

**Comment by drstevens** [comment]
The `accounts.google.com` and `firebaseapp.com` options are removed by turning off "Allow Google login buttons on third party sites"

![allow_google](https://user-images.githubusercontent.com/394663/119668005-c8074b80-be04-11eb-86a9-99f00d526c5f.png)

**Comment by nisc** [comment]
> The `accounts.google.com` and `firebaseapp.com` options are removed by turning off "Allow Google login buttons on third party sites"
> 
> ![allow_google](https://user-images.githubusercontent.com/394663/119668005-c8074b80-be04-11eb-86a9-99f00d526c5f.png)

I mean this one makes sense to me. Could have been better explained from a UI perspective, but that's what you sign up for if you check this option.

I still have hope that this whole thing is just some UI confusion, not an actual security problem. Would be great to get some update from whoever is looking into this.

**Comment by lllusion303** [comment]
> The `accounts.google.com` and `firebaseapp.com` options are removed by turning off "Allow Google login buttons on third party sites"
> 
> ![allow_google](https://user-images.githubusercontent.com/394663/119668005-c8074b80-be04-11eb-86a9-99f00d526c5f.png)

Tnx. I wondered why they disappeared after I'd finished configuring settings in a new profile.

**Comment by bongochong** [comment]
I ran into this recently and thought I was going insane, until I found this issue report. At some point over the past week this entry showed up under the _Sites that can always use cookies_ section:  
![allow_reuters.com](https://i.ibb.co/JtjZYg1/Screenshot-from-2021-05-26-15-24-35.png)  
And it simply cannot be removed. I cleared all site settings, cleared all cookies, and I still cannot delete the entry for reuters.com. To make matters worse, after I added the following entries to the _Always clear cookies when windows are closed_ section:  
![clear_reuters](https://i.ibb.co/jLdcksW/Screenshot-from-2021-05-26-15-25-04.png)  
Cookies from reuters.com will **still** persist across sessions, now requiring manual removal every time. There now appears to be no way to remove the allow list entry for cookies from reuters.com, and no way to have cookies from the site cleared on exist any more. Seeing that this is happening with other sites, on other installations, it is actually a pretty serious security issue.

**Comment by Tonev** [comment]
I just updated to the latest stable version, and like I usually do, I went through the settings to check if something I don't want has appeared over there. Unfortunately, I can confirm the issue from the original post is still present.

I tried a workaround that I thought might work, as it made some sense to me. I tried adding a website from the list of websites that can **always use cookies** to the list of websites that can **never use cookies**, hoping my approach will remove the website from the first list, as I don't understand how a website can always use cookies and can never use cookies at the same time, but unfortunately the same website ended up in both lists, which is interesting.

**Comment by Izofeu** [comment]
> I just updated to the latest stable version, and like I usually do, I went through the settings to check if something I don't want has appeared over there. Unfortunately, I can confirm the issue from the original post is still present.
> 
> I tried a workaround that I thought might work, as it made some sense to me. I tried adding a website from the list of websites that can **always use cookies** to the list of websites that can **never use cookies**, hoping my approach will remove the website from the first list, as I don't understand how a website can always use cookies and can never use cookies at the same time, but unfortunately the same website ended up in both lists, which is interesting.

https://github.com/brave/brave-browser/issues/12375#issuecomment-774566323

**Comment by Tonev** [comment]
Thank you for the trick!

I had to clear the browsing data for "Site and Shields Settings" in order to test another issue (https://github.com/brave/brave-browser/issues/15347), which in return cleared all websites from _brave://settings/cookies_ so guess that approach is yet another temporary solution until developers commit a fix.

**Comment by Michael82548** [comment]
> The `accounts.google.com` and `firebaseapp.com` options are removed by turning off "Allow Google login buttons on third party sites"
> 
> ![allow_google](https://user-images.githubusercontent.com/394663/119668005-c8074b80-be04-11eb-86a9-99f00d526c5f.png)

That actually worked for those two! Thank you!

I have also found that going to a website that is stuck and re-enabling the shields removes the cookie from the list

These are the only ones that remain say that they are embedded but I CANNOT remove those even if I do re-enable the shields. However, if I do end up going to those websites and disabling the shields, the icon for me changes to a reddit icon :\

There is probably a reason for this but I cannot find it because the UI is impossible to get hints from. Hence it taking this long to pin this problem down in the first place

**Comment by Michael82548** [comment]
Shields disabled on the embedded site
![image](https://user-images.githubusercontent.com/53535277/119897463-71863400-bf0e-11eb-8236-dc33d25bd2d5.png)

Shields enabled on the embedded site
![image](https://user-images.githubusercontent.com/53535277/119897553-97133d80-bf0e-11eb-97b8-42facedc0b4c.png)

I can't remove this one and I have NO INFO on what is preventing it from being removed. Brave's features probably weren't integrated well with the actual chromium backend so I cannot easily find the cause of this issue

**Comment by Michael82548** [comment]
Does brave store settings in the cookie? Is that why its getting added to that cookie list? That's confusing because when it is added to that list, the actual cookie is leaked when 3rd party cookies are enabled.

TLDR to this entire thing: Disabling shields on a website allows any website to read the cookie even if 3rd party cookies are enabled. This is leaking cookies and it is a security risk

One way to protect yourself would to be to re-enable shields every time you leave the website

This only affects SOME cookies. I have no idea what other brave settings could be blocking the deletion of cookies (Because it's not telling me), but this is NOT a good way to store settings in a browser if that is what it is doing, because its leaking website's cookies that seem to have certain brave features enabled/disabled

**Comment by Michael82548** [comment]
The fix for this would be to NOT store brave parameters in cookie settings so they don't leak or cause trouble! Fixing this issue would solve all of this if the cookies don't leak and abide by the 3rd party cookie setting!

**Comment by Michael82548** [comment]
> **SOLUTION** on Brave Browser Linux.
> Go to website in question and toggle the "Shields" to "UP" as opposed to "DOWN"
> Try it on github, you'll see it added or removed.
> This is one hell of a way for this to operate.
> Brave, give the user some control over this, make it more user friendly, and absolutely make it more easily removable.
> The default DOWN should be temporary, maybe clear cookies when windows are closed.
> There should be an UP, DOWN, etc., maybe a WTF as I can't believe this polar thinking!

Just realized that someone mentioned this already. Its also a solution on Microsoft Windows

Only to some cookies. Not all.

This has been mentioned 6 months ago

**Comment by Izofeu** [comment]
> > Still a problem in Version 1.19.92 Chromium: 88.0.4324.152 (Official Build) (64-bit)
> > Edit: I found a workaround to delete those entries - go to \AppData\Local\BraveSoftware\Brave-Browser\User Data\Default. Open file called Preferences and delete all site entries that you want to be gone. For example, if you want to delete a cookie permission from site "xxx.com", delete the following from the file (make sure the browser is closed):
> > `"xxx.com,*":{"expiration":"0","last_modified":"<somerandomnumbers>","model":0,"setting":2},`
> > Edit2: For sites that say "embedded on xxx.com" you might need to delete additional lines that look similar to the one above.
> 
> I have tried that with every value that had "myaccount.google.com" but they stayed.
> 
> [#12375 (comment)](https://github.com/brave/brave-browser/issues/12375#issuecomment-792388962)
> ^ This is probably the issue I am experiencing and it's very frustrating.

Does it say "Including third party cookies on this site" or does it say "embedded on <site>"? If the latter then you need to delete both the normal site and the embedded on site. Make sure you do this while your browser is closed too (check task manager, by default chromium browsers can stay open in background if there is something running in them).

**Comment by rebron** [comment]
I've filed a meta issue #16127 as there's a few related issues with cookie settings and shields. This particular issue has morphed into a couple different use cases and I don't want to dupe this issue out to https://github.com/brave/brave-browser/issues/11772 and #8842 as there's good discussions here for the workarounds. 

We are looking at this now and anticipate a fix in the next couple months. The meta issue has our punch list.

**Comment by nisc** [comment]
> We are looking at this now and anticipate a fix in the next couple months. The meta issue has our punch list.

Hello and thank you for prioritizing this. If you expect the fix to still take a few months from now, it would be helpful to have some statement/synthesis from the dev team on what risks users are exposing themselves to by using the current version, and which measures are needed to mitigate those risks.

**Comment by KevinSource0** [comment]
This workaround helps https://github.com/brave/brave-browser/issues/12375#issuecomment-848770755.
> The accounts.google.com and firebaseapp.com options are removed by turning off "Allow Google login buttons on third party sites"

However, for a privacy focused browser, why cookies from privacy hogging companies(google/facebook etc.) are enabled by default? At least if this cookies is not deletable for some reason, instead of failing silently, the browser should inform user why this is happening and direct user to the right place.

**Comment by snowbound** [comment]
Another cookie  deletion issue with Brave was solved for me by using a third party extension after 14+ months  https://github.com/brave/brave-browser/issues/9085#issuecomment-864297911

**Comment by RealDavidoff** [comment]
23 June 2021: I confirm the issue told by the OP still exists in Brave browser, unfortunately.
![image](https://user-images.githubusercontent.com/11789004/123095550-1587d000-d426-11eb-9dad-a418cf5100d7.png)

The first two are apps I use, and Brave offers 3dot options.
The last two I don't even know how they got there, and they have only the delete button (good) but it doesn't do a thing.

This issue existed ever since I installed Brave, but I didn't have time to report it sooner.

When you are at it, PLEASE make SITE-SPECIFIC COOKIE DELETION a click of a button!
It's not "Advanced", it's no need to scroll forever, it's no need for any hurdle at all: It's the most essential function  of any browser for conscious net citizens.

And because, other than this, I like Brave and WOULD continue using it. But not getting rid of site specific cookies AT WILL annoys me massively.

Thanks for reading :)

**Comment by RealDavidoff** [comment]
Oh, and seeing some other comments here: For me, NOTHING GOOGLE is allowed. I hate google. Far more dangerous than Microsoft. Microsoft has conscience. Google not. Google is EVIL.

**Comment by tneo** [comment]
This is stupid for a browser claiming to focus on privacy. Allow me to delete all cookies always unless I specify otherwise!

**Comment by Cautious-C** [comment]
This is stupid for a browser claiming to focus on privacy. 
Could not agree more. 
Some of these entries, google related ones, appeared automagically  ... has Google compromised your security features?
.. just about to export all my bookmarks and delete Brave from my system.

**Comment by pitsi** [comment]
Serious question.
Am I the only one that has NO google related sites under there? All FOUR sites I have there are mentioned in issue #15122, if anyone wants to have a look.

**Comment by Cautious-C** [comment]
When you say NOT google related..Do you mean it is not a Google domain name?
I suspect some of mine were google affiliated organisations, even if only using Google ad services...

**Comment by pitsi** [comment]
You can check all 4 sites I have there on the other issue report, I even have screenshots. They are xda-developers, no-ip, a page I run locally on a different pc and a site that I do not want to mention publically. None of the 4 is related to google.

**Comment by nisc** [comment]
Any update from the devs on this? At least a plan? Or an assessment of what this is? Sigh.

**Comment by Cautious-C** [comment]
Looks like one of my comments never appeared... If you set 
Block all cookies AND Clear cookies and site data when you quit Brave.
Then restart. It will delete all the entries. 
At least it worked for me!

**Comment by Michael82548** [comment]
> Looks like one of my comments never appeared... If you set
> Block all cookies AND Clear cookies and site data when you quit Brave.
> Then restart. It will delete all the entries.
> At least it worked for me!

Many users would not want to do that. it's also not reassuring that the same issue could arise at seemingly any given moment... or just never touch brave sheilds?

**Comment by Michael82548** [comment]
![image](https://user-images.githubusercontent.com/53535277/125888807-21d37f00-4953-44b0-9c56-5cca17dc9bc6.png)
Status as of Version 1.27.102 Chromium: 91.0.4472.124 (Official Build) beta (64-bit)

**Comment by nisc** [comment]
This is the thread that I show people when they ask me about my opinion on Brave.

**Comment by snowbound** [comment]
@Michael82548 That may be related to https://github.com/brave/brave-browser/issues/16127 which appears to be queued up to be fixed. When exactly it will be fixed is blowing in the wind. Hopefully fixing that bug also fixes https://github.com/brave/brave-browser/issues/9085#issuecomment-864297911 which I have been battling since March 2020 and which I had to resort to an extension to work around.

@nisc My frustration level with Brave at times has been off the scale especially when some of the issues/shortcomings don't occur in other Chromium browsers.

**Comment by xanoni** [comment]
> @nisc My frustration level with Brave at times has been off the scale especially when some of the issues/shortcomings don't occur in other Chromium browsers.

Yup. Great idea conceptually, but the priorities shown in practice are inconsistent with the marketing. I went back to Firefox.

**Comment by lllusion303** [comment]
@nisc @snowbound  other than Firefox, what other browsers do you use/recommend? 
I have a hardened FF setup solely for FB. I use DDG on iOS. I'm looking for another general use browser on macOS. Brave is proving to be mostly talk with little substance.

**Comment by snowbound** [comment]
@lllusion303 I used Firefox for years as my default browser and suck by it. However, it seemed to have issues after using it for 12+ hours in that it slowed down. Not having sync to mobile with Brave is frustrating and because I need to get work done for my clients I am looking at MS Edge whose sync abilities to iOS is more feature rich.

**Comment by rebron** [comment]
@nisc and @snowbound To be clear with process, when issues are filed we do our best to get the most information to be able define the issue, get it to a reproducible state, narrow it down, and work for a fix. We look at duplicates and related issues, we look at available workarounds which may help some users for the time being and we hope for additional comments that help shape a fix. 

When a particular git issue turns into a conversation that belongs in a community forum and calls to fix it now or fix it yesterday, or go with Edge or Firefox or whatever browser -- in other words conversations that don't solve the problem, we will file the issue elsewhere and address the problem there. We have the meta issue filed, we have several developers looking at all these issues, and we're moving to address this issue and those in the meta one. Cookie related issues are complex ones.

For further feedback not related to this issue, I invite folks to https://community.brave.com/ 
Comments, sentiments, feedback we take to heart. We're in this to be the best browser possible. If you believe differently, please share that on our community forum. We like to keep git issues at a tactical level - define the issue, gather supporting info, the actual fix, and verification of the fix.

We'll keep this github issue open for progress and notification purposes.

**Comment by snowbound** [comment]
@rebron Understand what you are saying. I have to admit that I got more meaningful responsive feedback for issues that I have had with Brave over the years, by posting about issues here on Github than I have on Brave Community.  One reason why I rarely go to Brave Community anymore other than responding to watched threads of mine.

Hence my frustrations were already high when I posted my bug report back in 2020 on the cookie issue.

**Comment by nisc** [comment]
> @nisc and @snowbound To be clear with process, when issues are filed we do our best to get the most information to be able define the issue, get it to a reproducible state, narrow it down, and work for a fix. We look at duplicates and related issues, we look at available workarounds which may help some users for the time being and we hope for additional comments that help shape a fix.

This issue has been open since October 2020.... that was 9 months ago. People are CONCERNED when they see bugs like this, especially Brave users who are probably more privacy aware than the average WWW user.

In this thread there've been repeated asks for a quick status update. Why? Because people want to understand if they can keep using Brave until this is fixed. Have they been browsing with gaping vulnerabilities for almost a year? People are CONCERNED.

The lack of communication is not good.

**Comment by nisc** [comment]
> @nisc @snowbound other than Firefox, what other browsers do you use/recommend?

Can't get fired for recommending Firefox.

**Comment by tneo** [comment]
> This is stupid for a browser claiming to focus on privacy. Allow me to delete all cookies always unless I specify otherwise!

This issue is related to the shields implementation of Brave. When you visit the site that is mentioned in the list for which you can't remove the settings it turns out the shield was turned off. This in itself means that the shield is blocking functionality of a site. To remove the site settings turn the shield back on. 

In itself that process is fine, the issue is that the shield in off-state should not result in me being unable to delete cookie settings through settings. Why does the shield implementation have such influence on behaviour in the browser settings? 

When you track shield site specifics in a list (which I can manipulate) this should be sufficient. As Brave only needs to know when to turn off the shield as apparently the site was not working with the shield up. Record the URL and you're done, no need to not allow the user not to change settings for cookie. Or changing default Chromium behaviour.

**Comment by bsclifton** [comment]
Going to do some cleanup on comments here. @nisc thanks for sharing your opinion. @rebron is a Brave employee and has shared updates here. I've ran this issue by our security team and while it is important to fix, it hasn't bubbled up to be more important than some of the other projects that our employees (the ones who could solve this issue) have been working on

Let's keep conversation on-topic please

**Comment by bridiver** [comment]
The fundamental issue here is that shield cookie settings are generated because they are a combination of cookie settings and the shields up/down state for the site. Because of that, they can't be stored the same way that regular chromium cookie settings are stored and that complicates things when it comes to the chromium content settings UI, particularly in the case of "block 3p" because that cannot be expressed as a single content setting in chromium. Things become even more complicated if you set a cookie override in shields and in the chromium UI for the same site. This has not been a high priority issue because most people just use shield settings. We kept this UI so advanced users could add custom rules, but the UI was only designed to handle chromium cookie settings and work needs to be done to correctly handle shields generated cookie settings as well.

**Comment by nisc** [comment]
> @nisc thanks for sharing your opinion.

Never a problem, and thank you for the clean-up. Glad to see we're getting a bit of traction here. Happier bird now.


> @bridiver most people just use shield settings

What do you suggest for the annoying people who always gotta try all the knobs and buttons? Would it all be worry-free if we only used the shield settings and never touched the chrome cookie UI?

Is it safe to use the "advanced" shield settings?

For context, I just checked **and I have 143 pages (Page-Down key presses)** of entries under brave://settings/cookies?search=cookies just in my primary Brave profile (fyi and don't worry, I didn't press the button 143 times — instead I counted the first page and then looked of the # of "embedded on" and "including third party cookies to extrapolate). I tried deleting a sample randomly but 100% of clicks did nothing.

So, are just the below 3 Chrome UI screens not to be touched to ensure defined behavior? Anything else that shouldn't be touched?

**One:**
<img src="https://user-images.githubusercontent.com/141336/126254702-9f4ad865-54b6-4298-84f7-f5f7a0230337.png" alt="drawing" width="200"/>

**Two:**
<img src="https://user-images.githubusercontent.com/141336/126254159-fb281b6e-dbd7-48c3-8a4f-01fa2c66770a.png" alt="drawing" width="200"/>

**Three:**
<img src="https://user-images.githubusercontent.com/141336/126254189-a74eb9e6-7f9a-4f0f-a0c0-31c5b1b7d274.png" alt="drawing" width="200"/>

**Comment by fusionneur** [comment]
Issue still present on Version 1.27.111 Chromium: 92.0.4515.131 (Official Build) (64-bit) - Windows

**Comment by tortino** [comment]
to delete sites from this list I tried loading each site and enabling again brave shields. this way they are removed from the "Sites that can always use cookies" list.
however the is a site that has changed domain: when I load that site it is redirected, and I have no chance to enable shields for it.
is there no other way to remove it? maybe editing some config file, or some sqlite db?

**Comment by Izofeu** [comment]
> to delete sites from this list I tried loading each site and enabling again brave shields. this way they are removed from the "Sites that can always use cookies" list.
> however the is a site that has changed domain: when I load that site it is redirected, and I have no chance to enable shields for it.
> is there no other way to remove it? maybe editing some config file, or some sqlite db?

https://github.com/brave/brave-browser/issues/12375#issuecomment-774566323

**Comment by pitsi** [comment]
The forementioned workaround worked for me, but for 3 of my 4 sites that are there. The one that failed is xda-developers and that is because it instantly redirects me from
`xda-developers.com`
to 
`www.xda-developers.com`
so brave takes it as a different site and it shows the shields as enabled. 
If anyone has any idea, please comment.

**Comment by Izofeu** [comment]
> The forementioned workaround worked for me, but for 3 of my 4 sites that are there. The one that failed is xda-developers and that is because it instantly redirects me from
> `xda-developers.com`
> to
> `www.xda-developers.com`
> so brave takes it as a different site and it shows the shields as enabled.
> If anyone has any idea, please comment.
> 
> ---edit
> Same thing for my local address. The shields are enabled for `192.168.1.5` and for any page under that, like `192.168.1.5/ariang` or `192.168.1.5:9090`, but the entry under sites that can always use cookies remains.

https://github.com/brave/brave-browser/issues/12375#issuecomment-907785988
I literally wrote what you need to do above your comment and you failed to read that.

**Comment by pitsi** [comment]
@Izofeu 
I did read that, as I read any comment posted here.
The file you mention, which is on `~/.config/BraveSoftware/Brave-Browser/Default/Preferences` here in linux, is a SINGLE line 100kb file of endless text. Even if I open it on an editor like geany (equivalent to notepad++ for windows), searching for that specific line (let alone deleting it) is a hard task. 
So I avoided it and tried the method @tortino described, which worked fairly well for me.

**Comment by fusionneur** [comment]
On Windows I cleared the offending sites by editing the file 'Preferences' under:
C:\Users\<username>\AppData\Local\BraveSoftware\Brave-Browser\User Data\Profile 1

Below the keys from the json that I emptied:


```
{
    "gaia_cookie": {},
    "custom_handlers": {
        "ignored_protocol_handlers": []
    },
    "partition": {
        "per_host_zoom_levels": {
            "x": {}
        }
    },
    "profile": {
        "content_settings": {
            "exceptions": {
                "automatic_downloads": {},
                **"braveShields": {},**
                "client_hints": {},
                "cosmeticFiltering": {
                    "*,*": {
                        "expiration": "0",
                        "last_modified": "xxxxxxxxxxxxxxx",
                        "model": 0,
                        "setting": 2
                    },
                    "*,https://firstparty": {
                        "expiration": "0",
                        "last_modified": "13254527517311214",
                        "model": 0,
                        "setting": 2
                    }
                },
                "fingerprintingV2": {
                    "*,*": {
                        "expiration": "0",
                        "last_modified": "13274896723771237",
                        "model": 0,
                        "setting": 2
                    }
                },
                "httpUpgradableResources": {},
                "media_engagement": {},
                "notifications": {},
                **"referrers": {**
                    "*,*": {
                        "expiration": "0",
                        "last_modified": "13257391282982328",
                        "model": 0,
                        "setting": 2
                    }
                },
                "shieldsAds": {
                    "*,*": {
                        "expiration": "0",
                        "last_modified": "13254527517311425",
                        "model": 0,
                        "setting": 2
                    }
                },
                **"shieldsCookies": {**
                    "*,*": {
                        "expiration": "0",
                        "last_modified": "13257391282982429",
                        "model": 0,
                        "setting": 2
                    },
                    "*,https://firstparty": {
                        "expiration": "0",
                        "last_modified": "13257391282982371",
                        "model": 0,
                        "setting": 1
                    }
                },
                "site_engagement": {},
                "sound": {},
                "trackers": {
                    "*,*": {
                        "expiration": "0",
                        "last_modified": "13254527517311425",
                        "model": 0,
                        "setting": 2
                    }
                }
            }
        }
    },
    "web_apps": {
        "daily_metrics": {}
    }
}
```

**Comment by reddyshyam** [comment]
I am a new user to Brave and so far loving it but when I look at how the bugs are squashed and the amount of time it takes in general, I will say I am very much disappointed.  For an issue to be addressed if it takes an hour or more, don't know how the users will be loyal.

**Comment by sosinovitch** [comment]
I'm having the same issue with Version 1.30.87 Chromium: 94.0.4606.71 (Official Build) (x86_64) on my Mac OS 10.15.7.

**Comment by bitshark** [comment]
> to delete sites from this list I tried loading each site and enabling again brave shields. this way they are removed from the "Sites that can always use cookies" list. however the is a site that has changed domain: when I load that site it is redirected, and I have no chance to enable shields for it. is there no other way to remove it? maybe editing some config file, or some sqlite db?

**This is the exact same issue that I have.   Under "Customized Behaviors" / "Sites that can always use cookies" I cannot delete any of the entries.**  It turns out these are sites for which I had shields down.  I am seeing this issue on Windows 10 x64 , Brave Version 1.31.91 Chromium: 95.0.4638.69 (Official Build) (64-bit).

**The only way I've managed to delete the entries is to manually go to each site in the list in a new tab, turn shields back on for each site one by one**, and then they spontaneously disappear from the Brave cookie configuration list (perhaps because I have clear cookies on exit enabled?).  If the target domain has changed in the meantime (for example, **if the domain now has an HTTP 302 redirect to somewhere new, the entry cannot be deleted from the list.**

Hope this helps.

**Comment by pitsi** [comment]
Since this behavior is directly related to the deactivation of shields, is there a setting that shows the sites for which the user has disabled brave's shields?

**Comment by Dylan-86** [comment]
Still an open issue.

We need a way to simply remove the cookies, also for websites for which we had to turn the shields down. 

Sometimes I turn the shields down for a website just because it won't work with shields up, but then I won't need that website again in my life. It's annoying that I can't easily delete the cookies of that specific website after one year.

**Comment by bridiver** [comment]
> Still an open issue.
> 
> 
> 
> We need a way to simply remove the cookies, also for websites for which we had to turn the shields down. 
> 
> 
> 
> Sometimes I turn the shields down for a website just because it won't work with shields up, but then I won't need that website again in my life. It's annoying that I can't easily delete the cookies of that specific website after one year.

I fully agree this is a problem, but it's a complicated problem because 3rd party blocking for a specific site can't be expressed as a single chromium content setting and they are not persisted as chromium cookie settings, they are generated at runtime from shield settings. I think what we probably need is a separate section for shield settings that is independent from chromium cookie settings. It's a trade off because that could be confusing to some users, but seems like the best solution. wdyt @petemill ? We had something like this is the old browser-laptop settings

**Comment by Dylan-86** [comment]
> > Still an open issue.
> > We need a way to simply remove the cookies, also for websites for which we had to turn the shields down.
> > Sometimes I turn the shields down for a website just because it won't work with shields up, but then I won't need that website again in my life. It's annoying that I can't easily delete the cookies of that specific website after one year.
> 
> I fully agree this is a problem, but it's a complicated problem because 3rd party blocking for a specific site can't be expressed as a single chromium content setting and they are not persisted as chromium cookie settings, they are generated at runtime from shield settings. I think what we probably need is a separate section for shield settings that is independent from chromium cookie settings. It's a trade off because that could be confusing to some users, but seems like the best solution. wdyt @petemill ? We had something like this is the old browser-laptop settings

I totally understand the technical problem. 
But trust me, today it took me one hour to manually open each and every website of the list (which are not even hyperlinked, so I had to manually copy/paste each address) and turn the shields on for them.

There are 2 alternatives imho:
1. From that list you can click on a shield icon next to the website that will turn the shields up for that. As I've done this today, I could see that by doing this the website was removed from the whitelist. 
2. Have a "**put shields down just for this session**" option. So, while you're navigating the site, the shields are down. As you close the tab, shields up again for that website. A non-permanent shield-down option.

Note that today I could not remove some of the websites from the whitelist since they do not exist anymore/are now redirects to other websites ---> no way to open the original URL and turn the shields up.

Thanks!

**Comment by tneo** [comment]
Good suggestion from Dylan-86! As it helps navigate the issue, without separating out the different settings, which may lead to confusion.

**Comment by drstevens** [comment]
> There are 2 alternatives imho:
> 
> 1. From that list you can click on a shield icon next to the website that will turn the shields up for that. As I've done this today, I could see that by doing this the website was removed from the whitelist.
>  2. Have a "put shields down just for this session" option. So, while you're navigating the site, the shields are down. As you close the tab, shields up again for that website. A non-permanent shield-down option.

This long standing issue has certainly been frustrating me as well. With that said, this suggestion oversimplifies the problem and highlights why it's probably not been "fixed" yet. It's not really a "shields up"/"shields down" question. For example, I might have "shields up" for a domain, fingerprinting and trackers are blocked, but I might still allow all cookies for a particular domain. In that case, shields are up, but the domain would still appear in this list today.

**Comment by ryancwalsh** [comment]
I'm having this problem too and was about to post about this bug in the Community, but then _many_ other posts popped up. One linked me here to GitHub.

Others, such as these, contain no link to this GitHub issue:

- https://community.brave.com/t/cannot-delete-sites-from-sites-that-can-always-use-cookies/214242
- https://community.brave.com/t/a-workaround-the-issue-cannot-delete-sites-from-sites-that-can-always-use-cookies/267784
- https://community.brave.com/t/cant-delete-sites-from-sites-that-can-always-set-cookies/173227/2

which means that nobody there will know that it's a confirmed bug.

(But this bug has remained open for _15 months_, so maybe those posters wouldn't benefit much from this thread anyway.)

**Comment by snowbound** [comment]
I have given up on this and https://github.com/brave/brave-browser/issues/9085 is never being fixed. Brave is too busy seeking Web 3.0.  As I indicated in other posts here I have resorted to using a 3rd party extension "Cookie AutoDelete" to delete a certain Google cookie upon exiting the browser.

**Comment by Canowyrms** [comment]
Just ran into this issue myself. I skimmed through the conversation here, but if I'm restating an already-mentioned solution, sorry, I missed it.

As another user here mentioned, this is related to the sync chain. I had just installed the latest version of Brave for Win 10 on a fresh installation of Win 10. The entries under "Sites that can always use cookies" appeared after entering my sync code, and, you can probably tell where this is going, I couldn't remove them.

My solution was to:
1. Leave the sync chain.
2. Clear ALL browsing data - absolutely everything, even under the 'advanced' tab.
3. Exit Brave and wait a few seconds (ensure all processes shut down).
4. Open Brave and check the "Sites that can always use cookies" section to confirm it was empty (it was).
5. Start a new sync chain.
6. Went back to the cookies section and confirmed the entries didn't come back from the grave.
7. Copy the sync chain code into password manager like BitWarden for when I need to update my other devices.

I have yet to use the new sync code on my other devices (only two others, so not terribly inconvenient, thankfully), however, I reckon I will follow steps 1 through 3 on my other devices **before** I switch to the new sync chain code. Somewhat inconvenient, but easier (for me, anyway) than mucking around with settings files, and I don't care that I've lost some browsing data. 🤞 that the entries don't come back from the grave on my other devices..

**Comment by istiak101** [comment]
What I found is disabling any sites shield will add that domain to the list.

**Comment by thejoshuahenry** [comment]
How is this issue STILL not resolved? This is embarrassing, Brave, and it's not worth sticking with your browser over.

**Comment by snowbound** [comment]
Have given up on reporting issues with Brave myself. It gets frustrating after spending countless hours testing and providing feedback and hearing crickets. Really tiring shouting into the wind.

**Comment by bridiver** [comment]
At this point afaik the only cookie settings that cannot be deleted are the ones that are created by disabling shields on a site or the pref to allow google logins and that is by design. Deleting those settings would leave things in a confusing state where it looks like shields is down, but it's only partially down because 3p cookies are still blocked. The fix here is to show those settings as coming from another source (similar to settings created by extensions). To remove those settings you need to navigate to the site and re-enable shields. Nightly and beta have added shields to `brave://settings` to make it easier to remove those overrides and it will land in release soon.

**Comment by bridiver** [comment]
also the shields generated cookie settings should not show up as editable in the cookies UI (similar to how extension created settings works)

**Comment by bridiver** [comment]
If anyone has a case where settings that were not adding by disabling shields or the google login pref cannot be deleted then that's something we need to look into, but otherwise I think this issue can be closed and a new issue opened up for properly displaying those settings as non-editable

**Comment by snowbound** [comment]
Well, I posted this issue regarding Google-specific cookie deletion almost 2 years ago  https://github.com/brave/brave-browser/issues/9085. I haven't tried the normal deletion in Brave because I found a 3rd party extension that works to delete the cookie. Frankly, I am tired/frustrated spending hours testing and posting feedback regarding Brave issues and hearing crickets. I have resigned myself to putting up with the current quirks in Brave until my patience wears out and I switch browsers.

**Comment by Rob-Garner** [comment]
I have the same issue on the newest version of Brave. I looked at the HTML code associated with the trash can symbol. There is no code to execute, so clicking the icon does nothing. Not very trustworthy of the Brave team to be allowing this.

**Comment by bridiver** [comment]
> I have the same issue on the newest version of Brave. I looked at the HTML code associated with the trash can symbol. There is no code to execute, so clicking the icon does nothing. Not very trustworthy of the Brave team to be allowing this.

@Rob-Garner I'm not sure what you're looking at, but the problem is definitely not that code does not execute when you click on the trashcan. The problem is that it doesn't always correctly delete cookie settings that were generated by shields. I'm also not sure what's untrustworthy about this because they are your settings that you selected and they can be changed in the shields panel.
<img width="591" alt="image" src="https://user-images.githubusercontent.com/34129/205704817-cc9ae529-0bd5-4de6-90a3-0245bae61e6d.png">

**Comment by KioriSun** [comment]
I had this issue in v1.47.171 used the method of disable google login in brave://settings/socialBlocking and ti worked.

**Comment by Brave-Matt** [comment]
A very angry +1 here:
![image](https://user-images.githubusercontent.com/3299098/217657562-5c5ce6ee-c590-4b58-8833-4a446e7a28d6.png)

**Comment by DutchPete** [comment]
Today is 2023-03-19, 2½ years after this was 1<sup>st</sup> reported here, and this issue still HAS NOT BEEN RESOLVED !!!!!

This is a privacy issue of a browser that claims to be focused on privacy. Yeah, sure 🤣
This is beyond a joke, and all the workarounds that are mentioned should not be necessary, the Brave devs should have picked this **immediately**, not when it is convenient.

If it is only done when convenient, then stop marketing yourself as a privacy focused browser: you can't have your cake and eat it.

**Comment by pitsi** [comment]
Tbh, it has been some months since I gave up on this issue. As mentioned above, the issue is somehow related to the shields.
I did what I mentioned above for those 3 sites and for the 4th one I simply deleted it from the "shields down" category under `brave://settings/content/braveShields` and it disappeared from "sited that can always use cookies". 
I also removed any sites that were under the "shields up" category in there. As it seems, disabling and reenabling shields for one adds it in there.

**Comment by rocketphewl** [comment]
This is still an issue three plus years later.. as far as I'm concerned this is a data breach allowing cookies to track me and no way to stop it easily.. Chrome and/or Brave browsers are frauds...

**Comment by bridiver** [comment]
> This is still an issue three plus years later.. as far as I'm concerned this is a data breach allowing cookies to track me and no way to stop it easily.. Chrome and/or Brave browsers are frauds...

Site overrides that were created through the shields UI can be changed through the shields UI, but overrides like "block third-party cookies" cannot be expressed as a single setting in the chromium cookie settings UI so you will see two rules (one for block and one for allow) that have slightly different values to block third party cookies while still allowing first party cookies. If you want to change those settings you need to do it through the shields UI, but you also need to understand what you're looking at in the chromium cookie settings because one rule does not necessarily tell the whole story.  Using only the chromium cookie settings UI, most people would not be able to figure out how to add a site override to block 3p cookies on only that site and most people would also not understand why there is an "allow all" rule for the site when looking at a block 3p setting added by shields. If I set to brave.com to "allow all" and then back to "block cross-site", I end up with this
<img width="671" alt="image" src="https://user-images.githubusercontent.com/34129/229673452-a05c221e-b167-40f9-962f-8ac347d3071a.png">
and
<img width="680" alt="image" src="https://user-images.githubusercontent.com/34129/229673498-f5b5116e-be30-465d-835e-4aca6dd3233a.png">
and "block cross-site cookies on brave.com" can only be expressed with both of those settings


The issue here is that shields cookie settings cannot be always be managed through the chromium cookie settings UI, not that cookie settings can't be changed. The other issue is that once you set a shields cookie override for a site, you can change it (through shields UI), but setting it back to the default doesn't remove the site override, it just makes the override the same as the default setting. I say "cannot always" because we don't know exactly what sequence of events results in people not being able to delete the individual settings added by shields. I just deleted the brave.com override in the chromium cookie settings by deleting both settings and it worked fine. I'm not doubting that some people cannot remove them, but getting into that state is not easily reproducible. 

I agree this is confusing for users who are using the chromium cookie settings instead of the shields UI and tbh some of this is my fault because I wanted to keep the chromium cookie settings UI when other people wanted to just remove it and only manage cookie settings through shields. In the end I think they were right because the other issue is that adding cookie settings directly to the chromium cookie settings UI (which can only be removed through the chromium cookie settings UI) can result in settings that can't be expressed in the options provided by shields (allow, block 3p and block all).

**Comment by drstevens** [comment]
>Site overrides that were created through the shields UI can be changed through the shields UI

@bridiver The issue with the Shields UI is that you need to be on the site to change the settings. This is sometimes very difficult because those domains or hostnames have since setup 301 or 308 redirects making it nearly impossible to change settings for the site through the Shields UI. This is how I ended up following this issue actually. There was more than one site which I had added exceptions for in the past that were very difficult if not impossible to use Shields UI on at some point in the future.

**Comment by 424344** [comment]
@drstevens  Agree. I'm in the same situation with redirected sites.

**Comment by bridiver** [comment]
So after more discussion the tentative plan is to make two changes:
1. Setting a site override to the default value will remove the override. This seems pretty straightforward and obvious, but there were some reasons why previously wanted the ability to maintain the override if the default was changed. This seems like an edge case and not worth the added complexity. This solves the issue of confusing settings showing up in cookies that people are sometimes misinterpreting. 
2. Completely remove the ability to manually add chromium cookie settings and replace it with a page to manage shields cookie settings. This will allow us to properly represent shield cookie rules like block cross-site and will also eliminate any issues with chromium cookie settings that change the effective cookie setting to be something other than what shields says it is. As mentioned above the edge cases that can be added in chromium cookie settings but don't have an analog in shields cookie settings aren't really meaningful or useful with ephemeral/third-party storage partitioning. This will simplify things and provide a consistent view of cookie settings that always aligns exactly with the shields drop-down

**Comment by drstevens** [comment]
@bridiver Thanks for the update. Love the browser and appreciate all that the Brave team does.
❤️

**Comment by edelstone** [comment]
I found something that worked for me on the MacOS Brave browser (at least regarding deleting the sites).

The same list of sites that is present in `brave://settings/cookies` is also present in `brave://settings/content/braveShields`.

The latter is where I was able to successfully remove sites. After doing so, all the sites were removed from both areas.

**Comment by j-seeley** [comment]
> Still a problem in Version 1.19.92 Chromium: 88.0.4324.152 (Official Build) (64-bit) Edit: I found a workaround to delete those entries - go to \AppData\Local\BraveSoftware\Brave-Browser\User Data\Default. Open file called Preferences and delete all site entries that you want to be gone. For example, if you want to delete a cookie permission from site "xxx.com", delete the following from the file (make sure the browser is closed): `"xxx.com,*":{"expiration":"0","last_modified":"<somerandomnumbers>","model":0,"setting":2},` Edit2: For sites that say "embedded on xxx.com" you might need to delete additional lines that look similar to the one above.

I tried this. The Preferences file has no mention of problem websites when the browser is closed.  Of course, when Brave is loaded, the Preferences file looks different.

**Comment by j-seeley** [comment]
> I had this issue in v1.47.171 used the method of disabling google login in brave://settings/socialBlocking and it worked.

This fixed it for me...  Go to brave://settings/socialBlocking and TOGGLE OFF all 4.

**Comment by ShivanKaul** [comment]
Is this still an issue now that the Google Login button is gone from social settings (and there is no dedicated social settings page)?

**Comment by kbulgrien** [comment]
Yes, I still cannot delete sites from, for example, Sites allowed to use third-party cookies.

**Comment by bridiver** [comment]
> Yes, I still cannot delete sites from, for example, Sites allowed to use third-party cookies.

The chromium cookie interface has been going through a lot of strange changes recently and we're thinking we just need to dump the upstream UI completely so we can handle these correctly. Chromium cookie settings are now split between cookies and site data and it's very confusing imo.

**Comment by asheroto** [comment]
Also having this issue. No console messages exist when clicking the trashcan icon.

**Comment by Brave-Matt** [comment]
+1 here https://community.brave.app/t/removing-cookies-for-mostly-all-websites-with-a-handful-of-exceptions/650257

Looks like this issue hasn't been updated in a while. Can we get another pass on this? cc @ShivanKaul

**Comment by ShivanKaul** [comment]
Some cookie settings for websites in brave://settings/cookies are controlled by Shields. Those can be removed by going to brave://settings/content/braveShields. Are there other issues? Can I have an STR?

<img width="760" height="189" alt="Image" src="https://github.com/user-attachments/assets/cd8aa19e-c033-492a-a77e-ccd3e7b7ef11" />
